### PR TITLE
feat(python): add DeltaTable.is_deltatable static method (#2662)

### PR DESCRIPTION
**Pull request by omkar-foss**
closes https://github.com/delta-io/delta-rs/issues/2662.

# Description
This adds a static method `is_deltatable(path, opts)` to the
`DeltaTable` Python class, which returns `True` if able to
locate a delta table at specified `path` and `False` otherwise.

It does so by reusing the Rust internal `is_delta_table_location()`
via the `DeltaTableBuilder`.

Additionally, this also adds documentation of the usage with
examples for the `DeltaTable.is_deltatable()` method.

### CODE REVIEW

**Comment by ion-elgreco** [comment]
@omkar-foss thanks for the PR, I do believe we should do a bit differently. Now we are loading the state of the table, which could be potentially very large. We could also simply look if _delta_log exists or not, and whether that is then also not empty.

**Comment by omkar-foss** [comment]
@ion-elgreco Yes I completely agree. I wasn't aware that the `DeltaTable()` constructor loads the entire Delta table eagerly (thanks for letting know!). If that's the case then indeed, checking the `_delta_log` and it's emptiness will be a lighter and a constant-time operation than checking the table state which might grow large and expensive.

As a matter of fact, the `IsDeltaTable` in Delta Lake OSS (Spark version) also checks for `_delta_log` and it's emptiness (see [here](https://github.com/delta-io/delta/blob/master/spark/src/main/scala/io/delta/tables/DeltaTable.scala#L809-L817)), so I suppose it should be perfectly appropriate to apply the same logic here as well. It'll provide consistency of functionality as well for Delta Lake users.

I'll update the PR accordingly, thanks for your quick review 😁👍🏽

**Comment by ion-elgreco** [comment]
@omkar-foss there might be some internal code on the rust side that does something like this already, I am not certain, but worth it to check

**Comment by omkar-foss** [comment]
Hey @ion-elgreco, I've updated the PR as discussed.

I've attempted to reuse the 
Rust side internal `is_delta_table_location()` (code [here](https://github.com/delta-io/delta-rs/blob/main/crates/core/src/logstore/mod.rs#L217-L231)) which reads the `_delta_log` and returns a boolean result, and it's working well.

Kindly check out the PR when you get some time and let me know if any improvement suggestions or changes required. Thanks!

File: crates/core/src/table/builder.rs [hunk 1]
```diff
@@ -330,6 +330,12 @@ impl DeltaTableBuilder {
         }
         Ok(table)
     }
+
+    /// Check if the [`DeltaTable`] exists
+    pub async fn verify_deltatable_existence(self) -> DeltaResult<bool> {
```
Comment (ion-elgreco) [inline crates/core/src/table/builder.rs:335]: Great work! Just minor feedback, I think this method could be on the DeltaTable struct directly

File: python/src/lib.rs [hunk 2]
```diff
@@ -136,6 +136,26 @@ impl RawDeltaTable {
...
+            builder = builder.with_storage_options(storage_options)
+        }
+        let res = rt()
+            .block_on(builder.verify_deltatable_existence())
+            .map_err(PythonError::from);
+        match res {
+            Ok(true) => Ok(true),
+            Ok(false) => Ok(false),
+            Err(err) => Err(err)?,
+        }
```
Comment (ion-elgreco) [inline python/src/lib.rs:156]: Can't we leave the match part out? 
```suggestion
        Ok(rt()
            .block_on(builder.verify_deltatable_existence())
            .map_err(PythonError::from)?)
```

File: crates/core/src/table/builder.rs (hunk 1 above)
Comment (omkar-foss) [reply crates/core/src/table/builder.rs]: Thanks! Indeed, having it in the `DeltaTable` struct makes more sense. Updated PR.

File: python/src/lib.rs (hunk 2 above)
Comment (omkar-foss) [reply python/src/lib.rs]: Done, updated PR.

**Comment by ion-elgreco** [comment]
@omkar-foss Nice work! Thanks for the contribution :)

**Review by ion-elgreco — APPROVED** [review]

**Pull request merged**